cmd/dmsgpost/dmsgpost.go

```diff
@@ -0,0 +1,226 @@
+// package main cmd/dmsgpost/dmsgpost.go
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	cc "github.com/ivanpirog/coloredcobra"
+	"github.com/skycoin/skywire-utilities/pkg/buildinfo"
+	"github.com/skycoin/skywire-utilities/pkg/cipher"
+	"github.com/skycoin/skywire-utilities/pkg/cmdutil"
+	"github.com/skycoin/skywire-utilities/pkg/logging"
+	"github.com/skycoin/skywire-utilities/pkg/skyenv"
+	"github.com/spf13/cobra"
+
+	"github.com/skycoin/dmsg/pkg/disc"
+	dmsg "github.com/skycoin/dmsg/pkg/dmsg"
+	"github.com/skycoin/dmsg/pkg/dmsghttp"
+)
+
+var (
+	dmsgDisc     string
+	dmsgSessions int
+	dmsgpostData string
+	//	dmsgpostHeader string
+	sk            cipher.SecKey
+	dmsgpostLog   *logging.Logger
+	dmsgpostAgent string
+	logLvl        string
+)
+
+func init() {
+	rootCmd.Flags().StringVarP(&dmsgDisc, "dmsg-disc", "c", "", "dmsg discovery url default:\n"+skyenv.DmsgDiscAddr)
+	rootCmd.Flags().IntVarP(&dmsgSessions, "sess", "e", 1, "number of dmsg servers to connect to")
+	rootCmd.Flags().StringVarP(&logLvl, "loglvl", "l", "", "[ debug | warn | error | fatal | panic | trace | info ]\033[0m")
+	rootCmd.Flags().StringVarP(&dmsgpostData, "data", "d", "", "dmsghttp POST data")
+	//	rootCmd.Flags().StringVarP(&dmsgpostHeader, "header", "H", "", "Pass custom header(s) to server")
+	rootCmd.Flags().StringVarP(&dmsgpostAgent, "agent", "a", "dmsgpost/"+buildinfo.Version(), "identify as `AGENT`")
+	if os.Getenv("dmsgpost_SK") != "" {
+		sk.Set(os.Getenv("dmsgpost_SK")) //nolint
+	}
+	rootCmd.Flags().VarP(&sk, "sk", "s", "a random key is generated if unspecified\n\r")
+	var helpflag bool
+	rootCmd.SetUsageTemplate(help)
+	rootCmd.PersistentFlags().BoolVarP(&helpflag, "help", "h", false, "help for "+rootCmd.Use)
+	rootCmd.SetHelpCommand(&cobra.Command{Hidden: true})
+	rootCmd.PersistentFlags().MarkHidden("help") //nolint
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "dmsgpost",
+	Short: "dmsgpost",
+	Long: `
+	┌┬┐┌┬┐┌─┐┌─┐┌─┐┌─┐┌─┐┌┬┐
+ 	 │││││└─┐│ ┬├─┘│ │└─┐ │
+	─┴┘┴ ┴└─┘└─┘┴  └─┘└─┘ ┴ `,
+	SilenceErrors:         true,
+	SilenceUsage:          true,
+	DisableSuggestions:    true,
+	DisableFlagsInUseLine: true,
+	Version:               buildinfo.Version(),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		if dmsgDisc == "" {
+			dmsgDisc = skyenv.DmsgDiscAddr
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		if dmsgpostLog == nil {
+			dmsgpostLog = logging.MustGetLogger("dmsgpost")
+		}
+		if logLvl != "" {
+			if lvl, err := logging.LevelFromString(logLvl); err == nil {
+				logging.SetLevel(lvl)
+			}
+		}
+
+		ctx, cancel := cmdutil.SignalContext(context.Background(), dmsgpostLog)
+		defer cancel()
+
+		pk, err := sk.PubKey()
+		if err != nil {
```

Comment:
Should not fail silently. I think we should just exit with error here if there is an error parsing or generating the key.

Reply:
again it's something copied from the dmsgget implementation